### PR TITLE
[feat]: [PL-26203]: adding Secret Manager Template DTO's and evaluator

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Describe your changes
Added DTO's and yaml and secret evaluator for SecretManager Script.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36545)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 8 Issues 
* Code issue 8 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36545)

File: 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@TypeAlias("io.harness.engine.expressions.ShellScriptBaseDTO")
+public class ShellScriptBaseDTO {
+  @ApiModelProperty(required = true) @NotNull @EntityIdentifier String orgIdentifier;
```
Comment (meenakshi-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java:25]: Remove NotNull check from here, also remove `required = true`

File: 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  @ApiModelProperty(required = true) @NotNull @EntityIdentifier String projectIdentifier;
```
Comment (meenakshi-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java:27]: same as above org/projectId are optional

File: 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  @ApiModelProperty(required = true) @NotNull @EntityIdentifier String identifier;
+
+  @ApiModelProperty(required = true) @NotNull @NGEntityName String name;
+
+  @ApiModelProperty(required = true) @NotNull String versionLabel;
+
+  @JsonProperty("type") String type;
```
Comment (meenakshi-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseDTO.java:35]: type should be enum TemplateType not String

File: 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseSource.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+@Data
+@Builder
+@JsonTypeName("Inline")
+@OwnedBy(PL)
+@TypeAlias("io.harness.engine.expressions.ShellScriptBaseSource")
+public class ShellScriptBaseSource {
+  @NotNull String type;
+
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> script;
```
Comment (meenakshi-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptBaseSource.java:31]: Should this be also marked as NotNull?

File: 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlDTO.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@Builder
+@TypeAlias("ShellScriptYamlDTO")
+public class ShellScriptYamlDTO implements YamlDTO {
+  @JsonProperty("connector") ShellScriptBaseDTO shellScriptBaseDTO;
```
Comment (meenakshi-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlDTO.java:21]: Why JsonProperty here is `connector` ?

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 40 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zSKOX4Wk4ENzanmlx:-N9zSKOX4Wk4ENzanmly:bw49oiq) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L40)):*
> ```Java
>     super.initialize();
>     // Add Shell Script Yaml Expression Functor
>     addToContext("__yamlExpression",
> ```

shouldn't we use a constant here for hardcoded string?

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 49 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zSP8n7Y9-YkzUlDzL:-N9zSP8n7Y9-YkzUlDzM:b-p0otb4) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L49)):*
> ```Java
>   protected List<String> fetchPrefixes() {
>     ImmutableList.Builder<String> listBuilder = ImmutableList.builder();
>     return listBuilder.add("__yamlExpression").add("__yamlExpression.connector").addAll(super.fetchPrefixes()).build();
> ```

same as above, constant for hard coded strings?

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 43 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zSSAl58CFIJbKbXRJ:-N9zSSAl58CFIJbKbXRK:b-p0otb4) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L43)):*
> ```Java
>         ShellScriptYamlExpressionFunctor.builder().rootYamlField(getShellScriptYamlField()).build());
>     // Add secret functor
>     addToContext("secrets", new SecretFunctor(functorToken));
> ```

same as above, constant for hard coded strings?

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 55 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zSfvP5QtP25ulc4zz:-N9zSfvP5QtP25ulc5--:b-qy72wb) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L55)):*
> ```Java
>     try {
>       YamlField yamlField = YamlUtils.readTree(yaml);
>       return yamlField.getNode().getField("connector");
> ```

constants?

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 57 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zSi8o8wHiMta0LOve:-N9zSi8o8wHiMta0LOvf:b6oyote) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L57)):*
> ```Java
>       return yamlField.getNode().getField("connector");
>     } catch (IOException e) {
>       throw new InvalidRequestException("Not valid yaml passed.");
> ```

log the exception, also error message can be improved

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionFunctor.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zTmq95EiSEUWtvXkQ:-N9zTmq95EiSEUWtvXkR:b-twj8oi) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionFunctor.java#L23)):*
> ```Java
> 
> @Value
> @Builder
> ```

class owner details missing

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java` line 26 at r2](https://reviewable.io/reviews/harness/harness-core/36545#-N9zTroTDpZMNkK7E1KM:-N9zTroTDpZMNkK7E1KN:b-twj8oi) ([raw file](https://github.com/harness/harness-core/blob/7e06e72c41f2bc33e44589513eb784321aa3c93f/878-ng-common-utilities/src/main/java/io/harness/engine/expressions/ShellScriptYamlExpressionEvaluator.java#L26)):*
> ```Java
> @Value
> @Builder
> public class ShellScriptYamlExpressionEvaluator extends EngineExpressionEvaluator {
> ```

class owner details missing

**Pull request closed without merging**